core/bin/external_node/src/config/mod.rs

```diff
@@ -55,6 +60,14 @@ impl RemoteENConfig {
             .get_main_contract()
             .rpc_context("get_main_contract")
             .await?;
+        let base_token_addr = match client.get_base_token_l1_address().await {
+            Err(ClientError::Call(err)) if ErrorCode::MethodNotFound.code() == err.code() => {
+                // This is the fallback case for when the EN tries to interact
+                // with a node that does not implement the zks_baseTokenL1Address endpoint.
+                ETHEREUM_ADDRESS
+            }
```

Comment:
We need to have this fallback on `Web3Error::NotImplemented` as well.

Reply:
I don't think I quite follow you, you mean this match should also handle the `Web3Error::NotImplemented` error, right? If so, I've done it [here](https://github.com/matter-labs/zksync-era/pull/1031/commits/f544652953edfd66390240d5b294a9a4ba47b106).

Reply:
@popzxc 
Tagging for visibility.

Reply:
@popzxc could you see this?

Reply:
Sorry for the delay. Yeah, that will do. There is a bug in the API server implementation right now (we use an incorrect code for the `NotImplemented` error), but we'll fix that separately, and until then the approach you used should work. Thanks!

Reply:
Nice! thanks!